pkg/chartmuseum/server/multitenant/api.go

```diff
@@ -96,9 +95,20 @@ func (server *MultiTenantServer) deleteChartVersion(log cm_logger.LoggingFn, rep
 }
 
 func (server *MultiTenantServer) uploadChartPackage(log cm_logger.LoggingFn, repo string, content []byte, force bool) (string, *HTTPError) {
-	filename, err := cm_repo.ChartPackageFilenameFromContent(content)
+	var filename string
+
+	// Check to make sure the chart version is valid
+	_, err := cm_repo.ChartVersionFromStorageObject(storage.Object{
+		Content: content,
+		// Since we only need content to check for the chart version
+		// left the others fields to be default
+	})
+	if err != nil {
+		return filename, &HTTPError{http.StatusBadRequest, err.Error()}
+	}
+
+	filename, err = cm_repo.ChartPackageFilenameFromContent(content)
 	if err != nil {
-		// TODO: this is an error if invalid semver, making "EnforceSemver2" useless...
 		return filename, &HTTPError{http.StatusInternalServerError, err.Error()}
```

Comment:
While I get familiar with everything, should we return a `400` here and skip the call to `ChartVersionFromStorageObject` entirely?

Reply:
cool, fixed in 2f11e5029b735bc6a92a9fc566a732dfccfa9c01!